TestOTLPWriteHandler is flaky

```
2025-04-22T16:20:07.0195447Z === RUN   TestOTLPWriteHandler
2025-04-22T16:20:07.0195554Z === RUN   TestOTLPWriteHandler/NoTranslation
2025-04-22T16:20:07.0195952Z     write_test.go:474: 
2025-04-22T16:20:07.0196297Z         	Error Trace:	D:/a/prometheus/prometheus/storage/remote/write_test.go:491
2025-04-22T16:20:07.0196802Z         	            				D:/a/prometheus/prometheus/storage/remote/write_test.go:474
2025-04-22T16:20:07.0196945Z         	Error:      	Sample not found: 
2025-04-22T16:20:07.0197661Z         	            	expected: {[{__name__ test.counter} {foo.bar baz} {instance test-instance} {job test-service}] 1745338803239 10}
2025-04-22T16:20:07.0205898Z         	            	actual  : [{[{__name__ test.histogram_count} {foo.bar baz} {instance test-instance} {job test-service}] 1745338803238 10} {[{__name__ test.histogram_bucket} {foo.bar baz} {instance test-instance} {job test-service} {le 0}] 1745338803238 2} {[{__name__ test.histogram_bucket} {foo.bar baz} {instance test-instance} {job test-service} {le 1}] 1745338803238 4} {[{__name__ test.histogram_bucket} {foo.bar baz} {instance test-instance} {job test-service} {le 4}] 1745338803238 10} {[{__name__ test.histogram_bucket} {foo.bar baz} {instance test-instance} {job test-service} {le +Inf}] 1745338803238 10} {[{__name__ target_info} {host.name test-host} {instance test-instance} {job test-service}] 1745338803238 1} {[{__name__ test.counter} {foo.bar baz} {instance test-instance} {job test-service}] 1745338803238 10} {[{__name__ test.gauge} {foo.bar baz} {instance test-instance} {job test-service}] 1745338803238 10} {[{__name__ test.histogram_sum} {foo.bar baz} {instance test-instance} {job test-service}] 1745338803238 30} {[{__name__ test.histogram_bucket} {foo.bar baz} {instance test-instance} {job test-service} {le 2}] 1745338803238 6} {[{__name__ test.histogram_bucket} {foo.bar baz} {instance test-instance} {job test-service} {le 3}] 1745338803238 8} {[{__name__ test.histogram_bucket} {foo.bar baz} {instance test-instance} {job test-service} {le 5}] 1745338803238 12}]
2025-04-22T16:20:07.0206417Z         	Test:       	TestOTLPWriteHandler/NoTranslation
2025-04-22T16:20:07.0206580Z === RUN   TestOTLPWriteHandler/UnderscoreEscapingWithSuffixes
2025-04-22T16:20:07.0206687Z     write_test.go:474: 
2025-04-22T16:20:07.0207009Z         	Error Trace:	D:/a/prometheus/prometheus/storage/remote/write_test.go:491
2025-04-22T16:20:07.0207507Z         	            				D:/a/prometheus/prometheus/storage/remote/write_test.go:474
2025-04-22T16:20:07.0207650Z         	Error:      	Sample not found: 
2025-04-22T16:20:07.0208402Z         	            	expected: {[{__name__ test_counter_total} {foo_bar baz} {instance test-instance} {job test-service}] 1745338803239 10}
2025-04-22T16:20:07.0216729Z         	            	actual  : [{[{__name__ test_histogram_bucket} {foo_bar baz} {instance test-instance} {job test-service} {le 1}] 1745338803238 4} {[{__name__ test_histogram_bucket} {foo_bar baz} {instance test-instance} {job test-service} {le 3}] 1745338803238 8} {[{__name__ test_histogram_bucket} {foo_bar baz} {instance test-instance} {job test-service} {le 5}] 1745338803238 12} {[{__name__ test_histogram_bucket} {foo_bar baz} {instance test-instance} {job test-service} {le +Inf}] 1745338803238 10} {[{__name__ test_gauge} {foo_bar baz} {instance test-instance} {job test-service}] 1745338803238 10} {[{__name__ test_histogram_count} {foo_bar baz} {instance test-instance} {job test-service}] 1745338803238 10} {[{__name__ test_histogram_bucket} {foo_bar baz} {instance test-instance} {job test-service} {le 0}] 1745338803238 2} {[{__name__ test_histogram_bucket} {foo_bar baz} {instance test-instance} {job test-service} {le 2}] 1745338803238 6} {[{__name__ test_histogram_bucket} {foo_bar baz} {instance test-instance} {job test-service} {le 4}] 1745338803238 10} {[{__name__ target_info} {host_name test-host} {instance test-instance} {job test-service}] 1745338803238 1} {[{__name__ test_counter_total} {foo_bar baz} {instance test-instance} {job test-service}] 1745338803238 10} {[{__name__ test_histogram_sum} {foo_bar baz} {instance test-instance} {job test-service}] 1745338803238 30}]
2025-04-22T16:20:07.0217221Z         	Test:       	TestOTLPWriteHandler/UnderscoreEscapingWithSuffixes
2025-04-22T16:20:07.0217376Z === RUN   TestOTLPWriteHandler/NoUTF8EscapingWithSuffixes
2025-04-22T16:20:07.0217480Z     write_test.go:474: 
2025-04-22T16:20:07.0217805Z         	Error Trace:	D:/a/prometheus/prometheus/storage/remote/write_test.go:491
2025-04-22T16:20:07.0218291Z         	            				D:/a/prometheus/prometheus/storage/remote/write_test.go:474
2025-04-22T16:20:07.0218434Z         	Error:      	Sample not found: 
2025-04-22T16:20:07.0219183Z         	            	expected: {[{__name__ test.counter_total} {foo.bar baz} {instance test-instance} {job test-service}] 1745338803239 10}
2025-04-22T16:20:07.0227531Z         	            	actual  : [{[{__name__ target_info} {host.name test-host} {instance test-instance} {job test-service}] 1745338803238 1} {[{__name__ test.gauge} {foo.bar baz} {instance test-instance} {job test-service}] 1745338803238 10} {[{__name__ test.histogram_sum} {foo.bar baz} {instance test-instance} {job test-service}] 1745338803238 30} {[{__name__ test.histogram_bucket} {foo.bar baz} {instance test-instance} {job test-service} {le 0}] 1745338803238 2} {[{__name__ test.histogram_bucket} {foo.bar baz} {instance test-instance} {job test-service} {le 4}] 1745338803238 10} {[{__name__ test.histogram_bucket} {foo.bar baz} {instance test-instance} {job test-service} {le +Inf}] 1745338803238 10} {[{__name__ test.counter_total} {foo.bar baz} {instance test-instance} {job test-service}] 1745338803238 10} {[{__name__ test.histogram_count} {foo.bar baz} {instance test-instance} {job test-service}] 1745338803238 10} {[{__name__ test.histogram_bucket} {foo.bar baz} {instance test-instance} {job test-service} {le 1}] 1745338803238 4} {[{__name__ test.histogram_bucket} {foo.bar baz} {instance test-instance} {job test-service} {le 2}] 1745338803238 6} {[{__name__ test.histogram_bucket} {foo.bar baz} {instance test-instance} {job test-service} {le 3}] 1745338803238 8} {[{__name__ test.histogram_bucket} {foo.bar baz} {instance test-instance} {job test-service} {le 5}] 1745338803238 12}]
2025-04-22T16:20:07.0228095Z         	Test:       	TestOTLPWriteHandler/NoUTF8EscapingWithSuffixes
2025-04-22T16:20:07.0228204Z --- FAIL: TestOTLPWriteHandler (0.00s)
2025-04-22T16:20:07.0228371Z     --- FAIL: TestOTLPWriteHandler/NoTranslation (0.00s)
2025-04-22T16:20:07.0228609Z     --- FAIL: TestOTLPWriteHandler/UnderscoreEscapingWithSuffixes (0.00s)
2025-04-22T16:20:07.0228820Z     --- FAIL: TestOTLPWriteHandler/NoUTF8EscapingWithSuffixes (0.00s)
```

if it's still there

also

```
2025-04-23T07:28:07.0471643Z --- FAIL: TestOTLPWriteHandler (0.00s)
2025-04-23T07:28:07.0484266Z     --- FAIL: TestOTLPWriteHandler/NoTranslation (0.00s)
2025-04-23T07:28:07.0485423Z         write_test.go:474: 
2025-04-23T07:28:07.0486203Z             	Error Trace:	/__w/prometheus/prometheus/storage/remote/write_test.go:491
2025-04-23T07:28:07.0486906Z             	            				/__w/prometheus/prometheus/storage/remote/write_test.go:474
2025-04-23T07:28:07.0487502Z             	Error:      	Sample not found: 
2025-04-23T07:28:07.0488462Z             	            	expected: {{0xc00011a980 ���������} 1745393287013 10}
2025-04-23T07:28:07.0489566Z             	            	actual  : [{{0xc00011af00 ���������} 1745393287012 10} {{0xc00011af00 ���������} 1745393287012 10} {{0xc00011af00 ��
2025-04-23T07:28:07.0492572Z �������} 1745393287012 10} {{0xc00011af00 �����������} 1745393287012 2} {{0xc00011af00 �����������} 1745393287012 6} {{0xc00011af00 �����������} 1745393287012 8} {{0xc00011af00 �����������} 1745393287012 12} {{0xc00011af00 ���������} 1745393287012 1} {{0xc00011af00 ���������} 1745393287012 30} {{0xc00011af00 �����������} 1745393287012 4} {{0xc00011af00 �����������} 1745393287012 10} {{0xc00011af00 �����������} 1745393287012 10}]
2025-04-23T07:28:07.0494521Z             	Test:       	TestOTLPWriteHandler/NoTranslation
2025-04-23T07:28:07.0495215Z     --- FAIL: TestOTLPWriteHandler/UnderscoreEscapingWithSuffixes (0.00s)
2025-04-23T07:28:07.0496126Z         write_test.go:474: 
2025-04-23T07:28:07.0496735Z             	Error Trace:	/__w/prometheus/prometheus/storage/remote/write_test.go:491
2025-04-23T07:28:07.0497499Z             	            				/__w/prometheus/prometheus/storage/remote/write_test.go:474
2025-04-23T07:28:07.0498092Z             	Error:      	Sample not found: 
2025-04-23T07:28:07.0498796Z             	            	expected: {{0xc00011aa80 ���������} 1745393287013 10}
2025-04-23T07:28:07.0499647Z             	            	actual  : [{{0xc00011b260 ���������} 1745393287012 10} {{0xc00011b260 ���������	�
2025-04-23T07:28:07.0502765Z             	            	�} 1745393287012 4} {{0xc00011b260 ���������	��} 1745393287012 8} {{0xc00011b260 ���������	��} 1745393287012 10} {{0xc00011b260 ���������} 1745393287012 1} {{0xc00011b260 ���������} 1745393287012 10} {{0xc00011b260 ���������} 1745393287012 10} {{0xc00011b260 ���������} 1745393287012 30} {{0xc00011b260 ���������	��} 1745393287012 2} {{0xc00011b260 ���������	��} 1745393287012 6} {{0xc00011b260 ���������	��} 1745393287012 10} {{0xc00011b260 ���������	��} 1745393287012 12}]
2025-04-23T07:28:07.0505163Z             	Test:       	TestOTLPWriteHandler/UnderscoreEscapingWithSuffixes
2025-04-23T07:28:07.0505872Z     --- FAIL: TestOTLPWriteHandler/NoUTF8EscapingWithSuffixes (0.00s)
2025-04-23T07:28:07.0506455Z         write_test.go:474: 
2025-04-23T07:28:07.0507004Z             	Error Trace:	/__w/prometheus/prometheus/storage/remote/write_test.go:491
2025-04-23T07:28:07.0507743Z             	            				/__w/prometheus/prometheus/storage/remote/write_test.go:474
2025-04-23T07:28:07.0508340Z             	Error:      	Sample not found: 
2025-04-23T07:28:07.0509007Z             	            	expected: {{0xc00011ac20 ���������} 1745393287013 10}
2025-04-23T07:28:07.0511056Z             	            	actual  : [{{0xc00086b3c0 ���������} 1745393287012 10} {{0xc00086b3c0 ���������} 1745393287012 10} {{0xc00086b3c0 ��	�������
2025-04-23T07:28:07.0512127Z             	            	��} 1745393287012 2} {{0xc00086b3c0 ��	�������
2025-04-23T07:28:07.0512762Z             	            	��} 1745393287012 4} {{0xc00086b3c0 ��	�������
2025-04-23T07:28:07.0513221Z             	            	�
2025-04-23T07:28:07.0514361Z �} 1745393287012 12} {{0xc00086b3c0 ���������} 1745393287012 1} {{0xc00086b3c0 ���������} 1745393287012 10} {{0xc00086b3c0 ���������} 1745393287012 30} {{0xc00086b3c0 ��	�������
2025-04-23T07:28:07.0515458Z             	            	��} 1745393287012 6} {{0xc00086b3c0 ��	�������
2025-04-23T07:28:07.0516076Z             	            	��} 1745393287012 8} {{0xc00086b3c0 ��	�������
2025-04-23T07:28:07.0516708Z             	            	��} 1745393287012 10} {{0xc00086b3c0 ��	�������
2025-04-23T07:28:07.0517270Z             	            	��} 1745393287012 10}]
2025-04-23T07:28:07.0517816Z             	Test:       	TestOTLPWriteHandler/NoUTF8EscapingWithSuffixes
2025-04-23T07:28:07.0518371Z FAIL
```

here https://github.com/prometheus/prometheus/actions/runs/14612347619/job/40992752704?pr=16472

```
--- FAIL: TestOTLPWriteHandler (0.00s)
    --- FAIL: TestOTLPWriteHandler/NoTranslation (0.00s)
        write_test.go:474: 
            	Error Trace:	/__w/prometheus/prometheus/storage/remote/write_test.go:491
            	            				/__w/prometheus/prometheus/storage/remote/write_test.go:474
            	Error:      	Sample not found: 
            	            	expected: {{0xc0001302e0          } 1745418679210 10}
            	            	actual  : [{{0xc000131100           	 } 1745418679209 10} {{0xc000131100           
            	            	 } 1745418679209 12} {{0xc000131100    
      } 1745418679209 1} {{0xc000131100          } 1745418679209 10} {{0xc000131100          } 1745418679209 30} {{0xc000131100            } 1745418679209 2} {{0xc000131100            } 1745418679209 6} {{0xc000131100            } 1745418679209 8} {{0xc000131100            } 1745418679209 10} {{0xc000131100          } 1745418679209 10} {{0xc000131100          } 1745418679209 10} {{0xc000131100            } 1745418679209 4}]
            	Test:       	TestOTLPWriteHandler/NoTranslation
    --- FAIL: TestOTLPWriteHandler/UnderscoreEscapingWithSuffixes (0.00s)
        write_test.go:474: 
            	Error Trace:	/__w/prometheus/prometheus/storage/remote/write_test.go:491
            	            				/__w/prometheus/prometheus/storage/remote/write_test.go:474
            	Error:      	Sample not found: 
            	            	expected: {{0xc000130640          } 1745418679210 10}
            	            	actual  : [{{0xc0001314c0    	 
            	            	     } 1745418679209 1} {{0xc0001314c0          } 1745418679209 10} {{0xc0001314c0          } 1745418679209 10} {{0xc0001314c0          } 1745418679209 30} {{0xc0001314c0            } 1745418679209 4} {{0xc0001314c0            } 1745418679209 6} {{0xc0001314c0            } 1745418679209 10} {{0xc0001314c0            } 1745418679209 12} {{0xc0001314c0            } 1745418679209 10} {{0xc0001314c0          } 1745418679209 10} {{0xc0001314c0            } 1745418679209 2} {{0xc0001314c0            } 1745418679209 8}]
            	Test:       	TestOTLPWriteHandler/UnderscoreEscapingWithSuffixes
    --- FAIL: TestOTLPWriteHandler/NoUTF8EscapingWithSuffixes (0.00s)
        write_test.go:474: 
            	Error Trace:	/__w/prometheus/prometheus/storage/remote/write_test.go:491
            	            				/__w/prometheus/prometheus/storage/remote/write_test.go:474
            	Error:      	Sample not found: 
            	            	expected: {{0xc000130b00          } 1745418679210 10}
            	            	actual  : [{{0xc0001317a0           	 } 1745418679209 10} {{0xc0001317a0     
     } 1745418679209 1} {{0xc0001317a0          } 1745418679209 10} {{0xc0001317a0          } 1745418679209 10} {{0xc0001317a0            } 1745418679209 6} {{0xc0001317a0            } 1745418679209 8} {{0xc0001317a0            } 1745418679209 10} {{0xc0001317a0          } 1745418679209 30} {{0xc0001317a0          } 1745418679209 10} {{0xc0001317a0            } 1745418679209 2} {{0xc0001317a0            } 1745418679209 4} {{0xc0001317a0            } 1745418679209 12}]
            	Test:       	TestOTLPWriteHandler/NoUTF8EscapingWithSuffixes
FAIL
```
https://github.com/prometheus/prometheus/actions/runs/14620735477/job/41019819510?pr=15238

--- a/storage/remote/write_test.go
+++ b/storage/remote/write_test.go
@@ -381,8 +381,8 @@ func TestWriteStorageApplyConfig_PartialUpdate(t *testing.T) {
 }
 
 func TestOTLPWriteHandler(t *testing.T) {
-	exportRequest := generateOTLPWriteRequest()
 	timestamp := time.Now()
+	exportRequest := generateOTLPWriteRequest(timestamp)
 	for _, testCase := range []struct {
 		name            string
 		otlpCfg         config.OTLPConfig
@@ -516,14 +516,12 @@ func handleOTLP(t *testing.T, exportRequest pmetricotlp.ExportRequest, otlpCfg c
 	return appendable
 }
 
-func generateOTLPWriteRequest() pmetricotlp.ExportRequest {
+func generateOTLPWriteRequest(timestamp time.Time) pmetricotlp.ExportRequest {
 	d := pmetric.NewMetrics()
 
 	// Generate One Counter, One Gauge, One Histogram, One Exponential-Histogram
 	// with resource attributes: service.name="test-service", service.instance.id="test-instance", host.name="test-host"
 	// with metric attribute: foo.bar="baz"
-
-	timestamp := time.Now()
 
 	resourceMetric := d.ResourceMetrics().AppendEmpty()
 	resourceMetric.Resource().Attributes().PutStr("service.name", "test-service")

